File: frontend/src/services/postService.js
import api from './api';

export const postService = {
  getAllPosts: () => api.get('/posts'),
  getPostById: (id) => api.get(`/posts/${id}`),
  getPostsByAuthor: (authorId) => api.get(`/posts/author/${authorId}`),
  createPost: (postData) => api.post('/posts', postData),
  updatePost: (id, postData) => api.put(`/posts/${id}`, postData),
  deletePost: (id) => api.delete(`/posts/${id}`),
};

File: mean/src/main/java/com/blog/mean/service/PostService.java
package com.blog.mean.service;

import com.blog.mean.dto.PostDTO;
import com.blog.mean.dto.CreatePostRequest;
import java.util.List;

public interface PostService {
    PostDTO createPost(CreatePostRequest request);
    PostDTO getPostById(Long id);
    List<PostDTO> getAllPosts();
    List<PostDTO> getPostsByAuthor(Long authorId);
    PostDTO updatePost(Long id, CreatePostRequest request);
    void deletePost(Long id);
}
